04-advanced-topics/network-analysis_rnaseq_01_wgcna.Rmd

```diff
@@ -420,13 +442,271 @@ readr::write_rds(bwnet,
+  )
+) +
+  ggforce::geom_sina() +
+  theme_classic()
+```
+
+This makes sense! 
+Looks like module 52 has elevated expression post treatment.
+
+## What genes are a part of module 52?
+
+If you want to know which of your genes make up a modules, you can look at the `$colors` slot. 
+This is a named list which associates the genes with the module they are a part of. 
+We can turn this into a data frame for handy use. 
+
+```{r}
+gene_module_key <- tibble::enframe(bwnet$colors, name = "gene", value = "module") %>%
+  # Let's add the `ME` part so its more clear what these numbers are and it matches elsewhere
+  dplyr::mutate(module = paste0("ME", module))
+```
+
+Now we can find what genes are a part of module 52. 
+
+```{r}
+gene_module_key %>%
+  dplyr::filter(module == "ME52")
+```
+
+Let's save this gene to module key to a TSV file for future use. 
+
+```{r}
+readr::write_tsv(gene_module_key,
+  file = file.path("results", "SRP133573_wgcna_gene_to_module.tsv")
+)
+```
+
+## Make a custom heatmap function
+
+We will make a heatmap that summarizes our differentially expressed module.
+Because we will make a couple of these, it makes sense to make a custom function for making this heatmap. 
+
+```{r}
+make_module_heatmap <- function(module_name,
+                                expression_mat = normalized_counts,
+                                metadata_df = metadata,
+                                gene_module_key_df = gene_module_key,
+                                module_eigengenes_df = module_eigengenes) {
+  # Create a summary heatmap of a given module.
+  #
+  # Args:
+  # module_name: a character indicating what module should be plotted, e.g. "ME52"
+  # expression_mat: The full gene expression matrix. Default is `normalized_counts`.
+  # metadata_df: a data frame with refinebio_accession_code and refinebio_treatment
+  #              as columns. Default is `metadata`.
+  # gene_module_key: a data.frame indicating what genes are a part of what modules. Default is `gene_module_key`.
+  # module_eigengenes: a sample x eigengene data.frame with samples as rownames. Default is `module_eigengenes`.
+  #
+  # Returns:
+  # A heatmap of expression matrix for a module's genes, with a barplot of the
+  # eigengene expression for that module.
+
+  # Set up the module eigengene with its refinebio_accession_code
+  module_eigengene <- module_eigengenes_df %>%
+    dplyr::select(module_name) %>%
+    tibble::rownames_to_column("refinebio_accession_code")
+
+  # Set up column annotation from metadata
+  col_annot_df <- metadata_df %>%
+    # Only select the treatment and sample ID columns
+    dplyr::select(refinebio_accession_code, refinebio_treatment) %>%
+    # Add on the eigengene expression by joining with sample IDs
+    dplyr::inner_join(module_eigengene, by = "refinebio_accession_code") %>%
+    # Store sample
+    tibble::column_to_rownames("refinebio_accession_code") %>%
+    # Arrange by treatment
+    dplyr::arrange(refinebio_treatment)
```

Comment:
Can you test something here? What if you include `refinebio_title` arrange by this as well I think this should put replicates together? Or see if just arranging  by accession number helps?
i.e. `dplyr::arrange(refinebio_treatment, refinebio_accession)`, or `dplyr::arrange(refinebio_treatment, refinebio_title)`

Reply:
Turns out replicates are in the order of accession IDs so I've used that for ordering.